src/ast/semantic_analyser.cpp

```diff
@@ -1926,10 +1926,16 @@ void SemanticAnalyser::visit(Tuple &tuple)
     Expression *elem = tuple.elems->at(i);
     elem->accept(*this);
 
+    // If elem type is none that means that the tuple contains some
+    // invalid cast (e.g., (0, (aaa)0)). In this case, skip the tuple
+    // creation. Cast already emits the error.
+    if (elem->type.IsNoneTy())
+      break;
```

Comment:
```suggestion
      return;
```

Looks like the if below can be dropped if this returns